File: cast/assignment3/src/main/java/com/castgroup/assignment3/service/PessoaService.java
package com.castgroup.assignment3.service;

import com.castgroup.assignment3.model.MensagemRespostaPessoa;
import com.castgroup.assignment3.model.Pessoa;

public interface PessoaService {

	public MensagemRespostaPessoa listarTodasPessoas();
	public MensagemRespostaPessoa buscarPessoaPorId(Long id);
	public MensagemRespostaPessoa gravarPessoa(Pessoa pessoa);
	public MensagemRespostaPessoa removerPessoa(Long id);
	
}

File: cast/assignment3/src/main/resources/static/js/app/PessoaService.js
'use strict'

angular.module('pessoa.services', []).factory('PessoaService', ["$http", "CONSTANTS", function($http, CONSTANTS) {

    var service = {};
    
    service.listarTodasPessoas = function() {
		return $http.get(CONSTANTS.listarTodasPessoas);
	}
    
    service.buscarPessoaPorId = function(idPessoa) {
		var url = CONSTANTS.buscarPessoaPorId + idPessoa;
		return $http.get(url);
	}
    
    service.gravarPessoa = function(pessoa) {
		return $http.post(CONSTANTS.gravarPessoa, pessoa);
	}
    
    service.removerPessoaPorId = function(idPessoa) {
		var url = CONSTANTS.removerPessoaPorId + idPessoa;
		return $http.delete(url);
	}
    
    return service;
}]);